src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -591,6 +590,142 @@ def get_default_realm(self) -> str:
+        all_realms_info = []
+        zonegroup_list = []
+        all_zonegroup_info = []
+        default_realm_id = ''
+        default_zonegroup_id = ''
+        rgw_realm_list_cmd = ['realm', 'list'] 
+        try:
+            _, out, err = mgr.send_rgwadmin_command(rgw_realm_list_cmd)
+            if 'realms' in out:
+                rgw_realm_list = out['realms']
+                default_realm_id = out['default_info']       
+        except SubprocessError as error:
+            raise DashboardException(error, http_status_code=500, component='rgw')
+        if len(rgw_realm_list) > 0:
+            for realm in rgw_realm_list:
+                cmd_realm_option = ['--rgw-realm', realm]
+                rgw_realm_info_cmd = ['realm', 'get']
+                rgw_realm_info_cmd += cmd_realm_option
+                try:
+                    _, out, err = mgr.send_rgwadmin_command(rgw_realm_info_cmd)
+                    if out:
+                        all_realms_info.append(out)
+                        if out['id'] == default_realm_id:
+                            defaults_info['default_realm'] = out['name']
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+        rgw_zonegroup_list_cmd = ['zonegroup', 'list'] 
+        try:
+            _, out, err = mgr.send_rgwadmin_command(rgw_zonegroup_list_cmd)
+            if out and 'zonegroups' in out:
+                zonegroup_list = out['zonegroups']
+                default_zonegroup_id = out['default_info']       
+        except SubprocessError as error:
+                raise DashboardException(error, http_status_code=500, component='rgw')
+        if len(zonegroup_list) > 0:
+            for zonegroup in zonegroup_list:
+                cmd_zonegroup_option = ['--rgw-zonegroup', zonegroup]
+                rgw_zonegroup_info_cmd = ['zonegroup', 'get']
+                rgw_zonegroup_info_cmd += cmd_zonegroup_option
+                try:
+                    _, out, err = mgr.send_rgwadmin_command(rgw_zonegroup_info_cmd)
+                    if out:
+                        all_zonegroup_info.append(out)
+                        if out['id'] == default_zonegroup_id:
+                            defaults_info['default_zonegroup'] = out['name']
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+        rgw_zone_get_cmd = ['zone', 'get'] 
+        try:
+            _, out, err = mgr.send_rgwadmin_command(rgw_zone_get_cmd)
+            if out:
+                defaults_info['default_zone'] = out['name']      
+        except SubprocessError as error:
+                raise DashboardException(error, http_status_code=500, component='rgw')
+        multisite_info.append(all_realms_info)
+        multisite_info.append(all_zonegroup_info)
+        multisite_info.append(defaults_info)
+        return multisite_info
```

Comment:
multisite_info looks like a struct instead of a list. 
```python
multisite_info = {
  "all_realms_info": ...,
  "all_zonegroup_info": ..,
  "defaults_info": ...
}
```

Reply:
This wasn't addressed.  Another way to deal with this could be to return a tuple `Tuple[List[RealmInfo], List[ZonegroupInfo], List[Defaults_info]]` or a class:
```python
class MultisiteInfo:
  realms_info...
  zongroup
  default_info...
```
It is hard to understand what the function will return without explicitly typing it (you have to read more code instead of the function header), and the list we are returning is semantically the same a class.

Reply:
Modified the code. Thanks @pereman2